code/languages/C++/flow-control/README.md

```diff
@@ -0,0 +1,203 @@
+
+Program statements that cause such jumps are called control statements. There are two major categories:
+
+* Loops
+* Decisions
+
+## Loops
+Loops repeat a statement a certain number of times, or while a condition is fulfilled. They are introduced by the keywords while, do, and for.
+
+### The while loop
+Its syntax is:
+
+``` 
+while (expression) statement
+```
+
+The while-loop simply repeats statement while expression is true. If, after any execution of statement, expression is no longer true, the loop ends, and the program continues right after the loop. For example, let's have a look at a countdown using a while-loop:
+``` C++
+// custom countdown using while
+#include <iostream>
+using namespace std;
+
+int main ()
+{
+  int n = 10;
+
+  while (n>0) {
+    cout << n << ", ";
+    --n;
+  }
+
+  cout << "liftoff!\n";
+}
+```
+### Output
+```
+10, 9, 8, 7, 6, 5, 4, 3, 2, 1, liftoff!
+```
+### The do-while loop
+
+A very similar loop is the do-while loop, whose syntax is:
+```
+do statement while (condition);
+```
+
+It behaves like a while-loop, except that condition is evaluated after the execution of statement instead of before, guaranteeing at least one execution of statement, even if condition is never fulfilled. For example, the following example program echoes any text the user introduces until the user enters goodbye:
+
+``` C++
+// echo machine
+#include <iostream>
+#include <string>
+using namespace std;
+
+int main ()
+{
+  string str;
+  do {
+    cout << "Enter text: ";
+    getline (cin,str);
+    cout << "You entered: " << str << '\n';
+  } while (str != "goodbye");
+}
+```
+### Output
+```
+Enter text: hello
+You entered: hello
+Enter text: who's there?
+You entered: who's there?
+Enter text: goodbye
+You entered: goodbye
+```
+## The for loop
+The for loop is designed to iterate a number of times. Its syntax is:
+```
+for (initialization; condition; increase) statement;
+```
+
+Like the while-loop, this loop repeats statement while condition is true. But, in addition, the for loop provides specific locations to contain an initialization and an increase expression, executed before the loop begins the first time, and after each iteration, respectively. Therefore, it is especially useful to use counter variables as condition.
+
+```C++
+// countdown using a for loop
+#include <iostream>
+using namespace std;
+
+int main ()
+{
+  for (int n=10; n>0; n--) {
+    cout << n << ", ";
+  }
+  cout << "liftoff!\n";
+}
+```
+
+### Output
+```
+10, 9, 8, 7, 6, 5, 4, 3, 2, 1, liftoff!
+```
+## Decision
+In this, we test a condition and based on whether the condition is true or false you execute a statement or set of statements. In C++ programming language, we can implement conditional branching using the following statements:
+
+* If….Else Statement
+* switch…case statement
+
+### if-else Statement
```

Comment:
You should also include else-if's here as well.

Reply:
I have made the following changes. Please review it.